crates/ruff/src/rules/pyupgrade/rules/use_pep585_annotation.rs

```diff
@@ -1,32 +1,35 @@
 use rustpython_parser::ast::Expr;
 
-use ruff_diagnostics::{AlwaysAutofixableViolation, Diagnostic, Fix};
+use ruff_diagnostics::{AutofixKind, Availability, Diagnostic, Fix, Violation};
 use ruff_macros::{derive_message_formats, violation};
 use ruff_python_ast::types::Range;
 
 use crate::checkers::ast::Checker;
 use crate::registry::AsRule;
 
-// TODO: document referencing [PEP 585]: https://peps.python.org/pep-0585/
 #[violation]
-pub struct DeprecatedCollectionType {
+pub struct NonPEP585Annotation {
     pub name: String,
+    pub fixable: bool,
 }
 
-impl AlwaysAutofixableViolation for DeprecatedCollectionType {
+impl Violation for NonPEP585Annotation {
+    const AUTOFIX: Option<AutofixKind> = Some(AutofixKind::new(Availability::Sometimes));
```

Comment:
NIt: Unrelated to this change: A `Availability::None` would remove the need for `Option<AutofixKind>`. It would then simply be `AUTOFX: AutofixKind = AutofixKind::NONE` (we could even inline `Availability` into `AutofixKind`